File: wave-compiler/examples/vector_add.py
from wave import kernel, f32, thread_id, barrier

@kernel
def vector_add(a: f32[:], b: f32[:], out: f32[:], n: u32):
    gid = thread_id()
    if gid < n:
        out[gid] = a[gid] + b[gid]

File: wave-compiler/examples/vector_add.rs
#[kernel]
fn vector_add(a: &[f32], b: &[f32], out: &mut [f32], n: u32) {
    let gid = thread_id();
    if gid < n {
        out[gid] = a[gid] + b[gid];
    }
}
